lib/hexpm_web/router.ex

```diff
@@ -51,7 +57,7 @@ defmodule HexpmWeb.Router do
   end
 
   scope "/", HexpmWeb do
-    pipe_through :browser
+    pipe_through [:browser, :login, :defaults]
```

Comment:
In another PR it would be nice to get all session behavior isolated into the Hexpm.Session module, but in an effort to keep this PR as small as possible this was not changed.

Reply:
Login has to do a db lookup and the same one HexpmWeb.Session does for the authenticated area I didn't want us to do that lookup twice. HexpmWeb.Session by default will perform redirects and so forth in case of expired and invalid sessions and not what we want for the general area vs the authenticated. I'd like to do more PRs to clean all this up.

Reply:
We have `plug :requires_login` for this: https://github.com/hexpm/hexpm/blob/main/lib/hexpm_web/controllers/dashboard_controller.ex#L4.

Don't we want to do the expired and invalid session checks everywhere we check if a user is logged in since you are logged in to the whole website not just the dashboard?

Reply:
@ericmj I hear what you're saying, think I just had a different mental model going on. So yes, we can adjust `Hexpm.Session `so it doesn't do redirects etc, replace the `:login` plug with HexpmWeb.Session and have one pipeline. Does this sound good?

Reply:
@ericmj yeah, I made a few changes and pushed them up, since it only took a minute. I think this is what you're after.